### PR TITLE
Aliases map for ibc denoms (hardcoded) in client.toml

### PR DESCRIPTION
**Pull request by pysel**
- add field
- modify config structure
- add aliases field to context
- rm
- add aliases to ctx

Closes: #https://github.com/osmosis-labs/osmosis/issues/3655
## Changes

Added map to `client.toml`, which maps human readable denom names (ex: uatom) to their ibc denom versions 

In order to get `aliases` map you should simply call client's `Context.Aliases`
## Documentation and Release Note

  - Does this pull request introduce a new feature or user-facing behavior changes? yes
  - Is a relevant changelog entry added to the `Unreleased` section in `CHANGELOG.md`? yes
  - How is the feature or change documented? changelog

### CODE REVIEW

**Comment by pysel** [comment]
I suppose there should also be added some functionality to dynamically generate this map?

**Comment by pysel** [comment]
@ValarDragon

**Comment by ValarDragon** [comment]
We can generate / populate the map osmosis-side!

File: client/config/toml.go [hunk 1]
```diff
@@ -26,6 +26,8 @@ output = "{{ .Output }}"
 node = "{{ .Node }}"
 # Transaction broadcasting mode (sync|async|block)
 broadcast-mode = "{{ .BroadcastMode }}"
+# Map of denom aliases
+aliases = { uatom = "ibc/27394FB092D2ECCD56123C74F36E4C1F926001CEADA9CA97EA622B25F41E5EB2", uakt = "ibc/1480B8FD20AD5FCAE81EA87584D269547DD4D436843C1D20F15E00EB64743EF4", ujuno = "ibc/46B44899322F3CD854D2D46DEEF881958467CDD4B3B10086DA49296BBED94BED" }
```
Comment (ValarDragon) [inline client/config/toml.go]: Can this be specified with multiple lines?

Or perhaps read more directly from a toml?

File: client/context.go [hunk 2]
```diff
@@ -387,3 +388,9 @@ func NewKeyringFromBackend(ctx Context, backend string) (keyring.Keyring, error)
 
 	return keyring.New(sdk.KeyringServiceName(), backend, ctx.KeyringDir, ctx.Input, ctx.KeyringOptions...)
 }
+
+func (ctx Context) WithAliases(aliases map[string]string) Context {
```
Comment (alexanderbez) [inline client/context.go]: WithAliases is too generic. Consider `WithIBCAliases`.

**Review by alexanderbez — CHANGES_REQUESTED** [review]

File: client/config/toml.go (hunk 1 above)
Comment (pysel) [reply client/config/toml.go]: yeah I also wanted to do that, but was getting an error

**Comment by pysel** [comment]
Screen Shot 2022-12-11 at 11 31 33

This is actually an illegal syntax in toml  ¯\_(ツ)_/¯

However it can be done like this:
Screen Shot 2022-12-11 at 11 33 49
I will try the second approach from osmosis-side

File: client/context.go (hunk 2 above)
Comment (ValarDragon) [reply client/context.go]: Should be denom aliases, applies equally to tokenfactory

**Comment by pysel** [comment]
@ValarDragon I do not think it is possible to do this from osmosis side. client.toml is being generated on sdk side in `ReadFromClientConfig`. 

We could: 
1) in the same function add functionality to dynamically add map structure to client.toml
2) somehow manage to do it from osmosis-side. I mean, it is possible I lied a little, but I do not know if that is the right way to do that
3) do not store it in client.toml (I still do not see the point of it honestly) and just add the map to `client.Context` when initializing root cmd
wdyt?

**Comment by ValarDragon** [comment]
can we in run-time read all keys for a toml object? (Instead of having a struct with all of them?)

**Comment by tac0turtle** [comment]
this feature was added to the sdk via bank metadata. it should be handled automatically now if the metadata is set

**Pull request closed without merging**